src/charm.py

```diff
@@ -138,6 +140,13 @@ def __init__(self, *args):
             charm=self,
             ports=[ServicePort(name="vault", port=self.VAULT_PORT)],
         )
+        self.ingress = IngressPerAppRequirer(
+            self,
+            port=self.VAULT_PORT,
+            strip_prefix=True,
+            scheme=lambda: "https",
+            redirect_https=True,
```

Comment:
Is this needed?

Reply:
Yes, this is needed to get an https url scheme.

Reply:
I'm unconvinced honestly, as far as I understand, this is to strip https call from the outside and transfer them into http in the inside of k8s. I may be wrong though.

Reply:
Is it not for redirecting `http` to `https`? In that case I would keep it, as it forces the use of `https`. But we should validate the behavior first since we seem to be having 3 different interpretations.

Reply:
redirect_https True is removed. We only need to keep scheme=lambda: "https" which is forcing the use of https.